rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java

```diff
@@ -2548,7 +2549,7 @@ private int determineParenthesisLevel(int childLineNumber, int parentLineNumber,
             return SINGLE_LINE_COMMENT;
         } else if (source.startsWith("/*", c)) {
             return MULTILINE_COMMENT;
-        } else if (source.startsWith("/", c)) {
+        } else if (source.startsWith("/", c) && validateIsDelimiter(node, c)) {
```

Comment:
So one thing that comes to mind is a `ConstantExpression` would be the only thing that could represent a "slashy" string. While a mathematical division should be a `BinaryExpression`. So wouldn't we be able to directly identify based on the type of the node whether it's one versus the other?

Reply:
My understanding is `BinaryExpression` could contain a `ConstantExpression` as the left or right expressions, meaning a slash can be found in `source` which belongs to a child node rather than the `BinaryExpression` we may be visiting now, hence the need to traverse the tree to find which node the `/` actually came from. Please let me know if I'm missing something.

Other thoughts were whether we should traverse just once and keep some kind of set of all `BinaryExpression` operators "skippable indexes" in the source for delimiters, but since I've only came across this issue for this particular case, I'm not sure if it's worth it.

Reply:
That's fair. I suppose we could encounter this -- albeit strange -- string concatenation:

```
/Hello, / + /world!/
```

Which would be a binary expression, but have both left and right being slashy strings.